### PR TITLE
Adopt ruff and address lint

### PR DESCRIPTION
**Pull request by blink1073**
Adopt ruff, which is faster and replaces flake8, isort, and pyupgrade
Move linters from pre-commit to Makefile
Add pip to dependabot
Address typings which were not being checked by pre-commit...

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -64,8 +64,12 @@ clean-env: ## Remove conda env
 	-conda env remove -n $(ENV) -y
 
 lint: ## Check code style
-	@pip install -q pre-commit
-	pre-commit run --all-files
+	@pip install -q -e ".[lint]"
+	@pip install -q pipx
+	ruff .
+	black --check --diff .
```
[the remark this answers is not in the document]
Comment (blink1073) [reply Makefile]: Done, thanks!

File: pyproject.toml [hunk 2]
```diff
@@ -126,22 +132,42 @@ exclude_lines = [
...
+# FBT001 Boolean positional arg in function definition
+"FBT001", "FBT002", "FBT003",
+# C901 `async_setup_kernel` is too complex (12)
+"C901",
+# C408 Unnecessary `dict` call (rewrite as a literal)
+"C408",
+# N806 Variable `reasonErr` in function should be lowercase
+"N806", "N802", "N803",
+# I252 Relative imports from parent modules are banned
+"I252",
```
Comment (kevin-bates) [inline pyproject.toml:160]: I'm seeing a warning for this entry when running `make lint`:
```
warning: `I252` has been remapped to `TID252`
```
```suggestion
# TID252 Relative imports from parent modules are banned
"TID252",
```
and is removed following the suggested change.

**Review by kevin-bates — COMMENTED** [review]
Thank you @blink1073 - this is great.

Just had the one suggestion to address the warning I'm seeing.

**Comment by kevin-bates** [comment]
Interesting that that `TID252` thing was a recent change.  Will approve and merge once the build completes.  Thank you!

**Review by kevin-bates — APPROVED** [review]

**Comment by blink1073** [comment]
I think we can expect a few paper cuts until ruff hits 0.1

**Pull request merged**